modeling/03_Per_Sat_Model.Rmd

```diff
@@ -0,0 +1,1047 @@
+  source("pySetup.R")
+} else {
+  use_condaenv(file.path(getwd(), "env"))
+}
+```
+
+And then import the needed modules
+
+```{python}
+import ee
+import os
+import time
+import pandas as pd
+import fiona
+```
+
+## GEE Setup
+
+In order to access the GEE API, we need to authenticate our identity. To do
+this, you must type `earthengine authenticate` in the terminal window. When
+your browser states 'You are now authenticated with the gcloud CLI', the
+authentication is complete. This authentication is valid for 7 days.
+
+Now, we need to initialize our GEE session.
+
+```{python}
+ee.Initialize(project = 'ee-ross-superior')
+```
+
+# Import assets
+
+## Load the labels file into the environment
+
+Because of how the later function is written, it works best to read these into R. I couldn't tell you why other than my function doesn't work with pandas dataframes.
+
+```{r}
+labels_file <- list.files('data/labels/', full.names = T)
+labels <- read_csv(labels_file)
+
+# subset for sen/ls
+sen_labels = labels %>% 
+  filter(mission == 'SEN2')
+ls_labels = labels %>% 
+  filter(mission != 'SEN2')
+
+# and for 5/7/8/9
+ls5_labels = labels %>% 
+  filter(mission == 'LS5')
+ls7_labels = labels %>% 
+  filter(mission == 'LS7')
+ls8_labels = labels %>% 
+  filter(mission == 'LS8')
+ls9_labels = labels %>% 
+  filter(mission == 'LS9')
+```
+
+### Quick QAQC of the labels
+
+We know there are some outliers in here. To create a good model, we'll want to remove any 'outliers' from each of the label sets. Because we're creating models for each mission, we'll perform these outlier clean ups for each mission. We'll define 'outliers' as the outer 0.02 and 0.98 percentiles. 
+
+Make some helper lists
+
+```{r}
+class_list = c("cloud", "other", "openWater", "lightNearShoreSediment",
+               "darkNearShoreSediment", "offShoreSediment")
+
+ls_band_list = c(expr(Red), expr(Green), expr(Blue), expr(B5), expr(B6), expr(B7))
+sen_band_list = c(expr(Red), expr(Green), expr(Blue), expr(B5), expr(B6), expr(B7), expr(B8), expr(B11), expr(B12))
+```
+
+#### Landsat 5
+
+Maybe a tidyfail, maybe a me problem, but we're creating new, very similar functions for each dataset. 
+
+```{r}
+qaqc_filter_ls5 <- function(band, cl) {
+  df <- ls5_labels %>% filter(class == cl)
+  values = df[{{ band }}] %>% as.vector(.)
+  df_sub = df %>% select(date, mission, lat, lon, class, vol_init, {{ band }})
+  cuts = quantile(values[[1]], c(0.02, 0.98), na.rm = T)
+  df_sub %>% 
+    filter(cuts[1] < {{ band }}, {{ band }} < cuts[2])
+}
+
+apply_qaqc_bands_ls5 <- function(bn) {
+  map2_dfr(c(bn, bn, bn, bn, bn, bn),
+           class_list,
+           qaqc_filter_ls5)
+}
+ls_5_filtered <-  map(ls_band_list, apply_qaqc_bands_ls5)
+
+ls_5_filt <- full_join(ls_5_filtered[[1]], ls_5_filtered[[2]]) %>% 
+  full_join(., ls_5_filtered[[3]]) %>% 
+  full_join(., ls_5_filtered[[4]]) %>% 
+  full_join(., ls_5_filtered[[5]]) %>% 
+  full_join(., ls_5_filtered[[6]]) %>% 
+  distinct()
```

Comment:
I think you might be able to simplify this piece if you want (and replace the other mission joining code with the same thing)

```suggestion
ls_5_filt <- ls_5_filtered %>% 
  purrr::reduce(full_join) %>%
  distinct()
```

Reply:
... so many functions not enough brain space. THANK YOU.